### DISCUSSION TITLE
[PSD3] Exercise 5.29

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/psd2223#130] [Esercizi (A-L)]
**Exercise 5.29** Express the following base 10 numbers in 16-bit fixed-point sign/magnitude format with eight integer bits and eight fraction bits. Express your answer in hexadecimal.

(a) $−13.5625$
(b) $42.3125$
(c) $−17.15625$

**Comment by Elia-Belli** [comment]
Convertiamo numeri decimali in numeri binari a 16bit in formato fixed-point sign/magnitude, con 8bit per la parte intera (segno compreso) e 8bit per la parte decimale: avendo specificato il formato non è necessario inserire il punto (una macchina non lo vedrebbe) ma lo tengo per leggibilità .

(a) $10001101.10010000_2 = 8D90_{16}$
(b) $00101010.01010000_2 = 2A50_{16}$
(c) $10010001.00101000_2 = 9128_{16}$

**Comment by FeddyLix17** [comment]
risposta alla soluzione di Elia
(a) in esadecimale dovrebbe essere 8D90
(b) in esadecimale dovrebbe essere 2A50
per il resto tutto corretto

**Comment by Elia-Belli** [comment]
Corretto, grazie
